lib/ViberMessageTypes.js

```diff
@@ -0,0 +1,158 @@
+module.exports = ViberMessageTypes = {
+    TextMessage: TextMessage,
+    PictureMessage: PictureMessage,
+    VideoMessage: VideoMessage,
+    FileMessage: FileMessage,
+    ContactMessage: ContactMessage,
+    LocationMessage: LocationMessage,
+    UrlMessage: UrlMessage,
+    StickerMessage: StickerMessage,
+    RichMediaMessage: RichMediaMessage
+};
+
+function TextMessage(text) {
+    return {
+        'receiver': '',
+        'min_api_version': 1,
+        'sender': {
+            'name': '',
+            'avatar': ''
+        },
+        'tracking_data': 'tracking data',
+        'type': 'text',
+        'text': text
+    };
+}
+
+function PictureMessage(description, mediaUrl, thumbnailUrl) {
+    let message = {
+        'receiver': '',
+        'min_api_version': 1,
+        'sender': {
+            'name': '',
+            'avatar': ''
+        },
+        'tracking_data': 'tracking data',
+        'type': 'picture',
+        'text': description,
+        'media': mediaUrl,
+    };
+
+    if (thumbnailUrl) { message[thumbnail] = thumbnailUrl; };
+
+    return message;
+}
+
+function VideoMessage(mediaUrl, size, duration, thumbnail) {
+    let message = {
```

Comment:
Why not factor Message ? Like This:
let message = {
'receiver': '',
        'min_api_version': 1,
        'sender': {
            'name': '',
            'avatar': ''
        }
}

And you will add some proprieties for the message in all function

Reply:
thanks! i'll fix this asap. Had no time in past few days at all